targeting_test.go

```diff
@@ -194,6 +194,40 @@ func TestStringTargeting(t *testing.T) {
 	testTargetingString(targetingProto.Targeting_STARTS_WITH, "d", "abc", t, false, false)
 	testTargetingString(targetingProto.Targeting_STARTS_WITH, "c", "abc", t, false, false)
 	testTargetingString(targetingProto.Targeting_STARTS_WITH, "", "abc", t, true, false)
+
+	testTargetingString(targetingProto.Targeting_SEMVER_EQUALS, "2.10.3", "2.10.3", t, true, false)
```

Comment:
As mentioned in Semver [doc](https://pkg.go.dev/golang.org/x/mod/semver), They support this format: vMAJOR[.MINOR[.PATCH[-PRERELEASE][+BUILD]]], so can you include PRERELEASE and BUILD in your Unit tests ?

Reply:
Done :+1:  I also added a check on the semver validity as we want the targeting to return false if one or both versions aren't valid.